base/abstractdict.jl

```diff
@@ -242,9 +242,8 @@ julia> keytype(Dict(Int32(1) => "foo"))
 Int32
 ```
 """
-keytype(::Type{AbstractDict{K,V}}) where {K,V} = K
+keytype(::Type{<:AbstractDict{K,V}}) where {K,V} = K
```

Comment:
For consistency with some of the other `eltype` calls, should we define this `@isdefined(K) ? K : Any`?